test/box-luatest/meta-temporary_spaces_test.lua

```diff
@@ -0,0 +1,440 @@
+            "options parameter 'temporary' should be of type boolean",
+            s.alter, s, { temporary = 'meta' }
+        )
+        t.assert_error_msg_contains(
+            "meta-temporariness cannot change",
+            _space.update,
+                _space, s.id, {{'=', 6, { temporary = 'meta' }}}
+        )
+
+        local s = box.space.metatemp
+        t.assert_error_msg_contains(
+            "meta-temporariness cannot change",
+            s.alter,
+                s, { temporary = true }
+        )
+        t.assert_error_msg_contains(
+            "meta-temporariness cannot change",
+            _space.update,
+                _space, s.id, {{'=', 6, { temporary = true }}}
+        )
+    end)
+end
+
+-- check which features aren't supported for meta-temporary spaces
+g.test_meta_temporary_dont_support = function()
+    g.server:exec(function()
+        local t = require 'luatest'
+
+        local s = box.schema.space.create('metatemp', { temporary = 'meta',
+                                                        id = 1111111111 })
+        s:format {{'k', 'unsigned'}}
+        s:create_index('pk')
+        box.schema.func.create('foo', {
+            is_sandboxed = true,
+            is_deterministic = true,
+            body = "function() end"
+        })
+        t.assert_error_msg_equals(
+            "meta-temporary space does not support functional indexes",
+            s.create_index, s, 'func', { func = 'foo' }
+        )
+
+        t.assert_error_msg_equals(
+            "Grant access to meta-temporary space 'metatemp' "
+            .. "is denied for user 'guest'",
+            box.schema.user.grant, 'guest', 'read', 'space', s.name
+        )
+    end)
+end
+
+-- check which sql features aren't supported for meta-temporary spaces
+g.test_meta_temporary_sql_dont_support = function()
+    g.server:exec(function()
+        local t = require 'luatest'
+        box.schema.space.create('metatemp', { temporary = 'meta',
+                                              id = 1111111112 })
+        box.schema.space.create('test', { format = { 'k', 'unsigned' } })
+
+        local function sql(stmt)
+            local ok, err = box.execute(stmt)
+            local _ = ok == nil and error(err)
+        end
+
+        t.assert_error_msg_contains(
+            "triggers are not supported for meta-temporary spaces",
+            sql, [[
+                create trigger "my_trigger" before insert on "metatemp"
+                for each row begin
+                    insert into "metatemp" values (1);
+                end;
+            ]]
+        )
+
+        t.assert_error_msg_contains(
+            "sequences are not supported for meta-temporary spaces",
+            sql, [[
+                alter table "metatemp"
+                    add column "k" integer primary key autoincrement
+            ]]
+        )
+
+        -- alter table works though
+        sql [[ alter table "metatemp" add column "k" int primary key ]]
```

Comment:
Please, call functions using `()`. The most relaxed way of calling is when we can do `func{opts}` instead of `func({opts})`. But otherwise always use `()`.

Reply:
done